### PR TITLE
Trig integrals

### PR DESCRIPTION
**Pull request by b-mcg**
I fixed a bug in trigintegrate that lead to an incorrect result when integrating cot(x)_cos(x)_*2.  I also updated two previous tests and added one for what used to be an incorrect result.  All integral tests passed.

``` python
sympy/integrals/tests/test_deltafunctions.py[2] ..                                                                                               [OK]
sympy/integrals/tests/test_failing_integrals.py[18] ffwwffffffwwffwffs                                                                           [OK]
sympy/integrals/tests/test_heurisch.py[27] .........f....wwww.........                                                                           [OK]
sympy/integrals/tests/test_integrals.py[108] .....ffff..www..........................................................................................
....                                                                                                                                             [OK]
sympy/integrals/tests/test_lineintegrals.py[1] .                                                                                                 [OK]
sympy/integrals/tests/test_manual.py[20] ....................                                                                                    [OK]
sympy/integrals/tests/test_meijerint.py[21] ......www............                                                                                [OK]
sympy/integrals/tests/test_prde.py[13] .............                                                                                             [OK]
sympy/integrals/tests/test_quadrature.py[14] ..............                                                                                      [OK]
sympy/integrals/tests/test_rationaltools.py[8] ........                                                                                          [OK]
sympy/integrals/tests/test_rde.py[10] ......f...                                                                                                 [OK]
sympy/integrals/tests/test_risch.py[32] ................................                                                                         [OK]
sympy/integrals/tests/test_singularityfunctions.py[1] .                                                                                          [OK]
sympy/integrals/tests/test_transforms.py[19] ...wwwwww..........                                                                                 [OK]
sympy/integrals/tests/test_trigonometry.py[4] ....                                                                                               [OK]

___________________________________________________________________ slowest tests ___________________________________________________________________
test_fourier_transform - Took 10.250 seconds
test_messy - Took 11.184 seconds
test_inverse_laplace_transform - Took 11.814 seconds
test_issue_10847 - Took 12.234 seconds
test_issue_4890 - Took 14.701 seconds
test_expint - Took 21.860 seconds
test_issue_10798 - Took 25.086 seconds
test_issue_10680 - Took 33.549 seconds
================================== tests finished: 258 passed, 22 skipped, 18 expected to fail, in 402.14 seconds ===================================
Out[2]: True
```

### CODE REVIEW

File: sympy/integrals/trigonometry.py [hunk 1]
```diff
@@ -79,8 +79,20 @@ def trigintegrate(f, x, conds='piecewise'):
...
+            # Make sure to choose the positive one
+            # otherwise an incorrect integral can occur.
+            if n < 0 and m > 0:
+                m_ = True
+                n_ = False
+            elif m < 0 and n > 0:
+                n_ = True
+                m_ = False
+
+            # Both n and m are odd and positive
```
Comment (jksuom) [inline sympy/integrals/trigonometry.py]: It seems that both n and m could be negative.

Comment (b-mcg) [inline sympy/integrals/trigonometry.py]: In that case it would just fall back to the original method.

Comment (b-mcg) [inline sympy/integrals/trigonometry.py]: I added a check for both being negative so the smaller n or m in abs value will be chosen.

**Comment by asmeurer** [comment]
Based on the test changes, the answers were correct before (`log(cos(x)**2 - 1)/2 + cos(x)**2/2` is the same as `log(sin(x)) - sin(x)**2/2` up to a constant), but the new answers are simpler, so I like the change.

**Comment by b-mcg** [comment]
Yeah, you're right.  The original answer is equal to the new one when its constant, `C`, is equal to `-(1/2)`.  I was trippin', my bad.  The new answers are simpler though.

**Comment by moorepants** [comment]
@asmeurer Is this ready to merge?

File: sympy/integrals/trigonometry.py [hunk 2]
```diff
@@ -79,8 +79,25 @@ def trigintegrate(f, x, conds='piecewise'):
...
+                n_ = True
+                m_ = False
+            # Both are negative so choose the smallest n or m
+            # in absolute value for simplest substitution.
+            elif (n < 0 and m < 0):
+                n_ = n > m
+                m_ = not (n > m)
+
+            # Both n and m are odd and positive
+            else:
```
Comment (gxyd) [inline sympy/integrals/trigonometry.py]: For this case only, I don't think we have test cases. Also when this case is reached, this is handled the same way as it used to be,
```
>>> n = m = -1
>>> trigintegrate(sin(x)**n*cos(x)**m, x)
-log(sin(x)**2 - 1)/2 + log(sin(x))
```

File: sympy/integrals/trigonometry.py [hunk 3]
```diff
@@ -79,8 +79,25 @@ def trigintegrate(f, x, conds='piecewise'):
...
+            # otherwise an incorrect integral can occur.
+            if n < 0 and m > 0:
+                m_ = True
+                n_ = False
+            elif m < 0 and n > 0:
+                n_ = True
+                m_ = False
+            # Both are negative so choose the smallest n or m
+            # in absolute value for simplest substitution.
+            elif (n < 0 and m < 0):
```
Comment (gxyd) [inline sympy/integrals/trigonometry.py]: This is the code block which is un-tested.
```
>>> n = m = -1
>>> trigintegrate(sin(x)**n*cos(x)**m, x)
-log(sin(x)**2 - 1)/2 + log(sin(x))
```
(Correct code-block)

**Comment by gxyd** [comment]
I'll merge this in 24 hours, if there are no objections and if it passes tests.

**Pull request merged**